fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/StructurePoolAddCallback.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A callback for newly added structure pools.
+ *
+ * <p><strong>Word of warning</strong>: Mods may be editing on the structure pool from user configured data packs
+ * instead of the builtin Minecraft or mod resources.
+ *
+ * <p>Example usage:
+ * <pre>{@code
+ * StructurePoolAddCallback.EVENT.register(structurePool -> {
+ * 	if (structurePool.getUnderlying().getId().toString().equals("minecraft:village/common/butcher_animals")) {
+ * 		structurePool.addStructurePoolElement(StructurePoolElement.method_30425("village/common/animals/pigs_1").apply(StructurePool.Projection.RIGID), 2);
+ *    }
+ * });}
+ * </pre>
+ */
+public interface StructurePoolAddCallback {
+	/**
+	 * Called when structure pools are reloaded at data pack reload time.
+	 */
+	Event<StructurePoolAddCallback> EVENT = EventFactory.createArrayBacked(StructurePoolAddCallback.class,
```

Comment:
imo this should be replaced with an `event(Identifier)` method that returns a corresponding event factory for the structure pool id. 
It is convenient for modders as there's no extra `if` statement, and it is also quite uncommon to be adding the same pool element to two different structure pools.
`ScreenEvents` and the registry events do this

Reply:
I mean, you probably will have to add your element to multiple structure pools with villages that have many structure pools for each biome they spawn in. It is also convenient for keeping your if checks in one event rather than having several events.